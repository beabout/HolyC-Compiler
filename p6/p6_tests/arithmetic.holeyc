void fn(int global_int){
  bool b;
  b = true;
  global_int = 1;
  int side_one;
  int side_two;
  side_one = global_int;
  side_two = global_int + 1;
  side_two = global_int - (global_int / 2);
  b = (true && !b);
  return;
}