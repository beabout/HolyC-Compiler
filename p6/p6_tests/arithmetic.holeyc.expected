[BEGIN GLOBALS]
[END GLOBALS]
[BEGIN fn LOCALS]
global_int (formal)
b (local)
side_one (local)
side_two (local)
tmp0 (tmp)
tmp1 (tmp)
tmp2 (tmp)
tmp3 (tmp)
tmp4 (tmp)
[END fn LOCALS]
lbl_fun_fn: enter fn
getarg 1 [global_int]
[b] := 1
[global_int] := 1
[side_one] := [global_int]
[tmp0] := [global_int] ADD64 1
[side_two] := [tmp0]
[tmp1] := [global_int] DIV64 2
[tmp2] := [global_int] SUB64 [tmp1]
[side_two] := [tmp2]
[tmp3] := NOT8 [b]
[tmp4] := 1 AND8 [tmp3]
[b] := [tmp4]
goto lbl_0
lbl_0: leave fn